libs/crypto/keys_manager_impl.cpp

```diff
@@ -36,23 +38,53 @@ namespace iroha {
    * @return keypair on success, otherwise nullopt
    */
   template <typename T, typename V>
-  auto deserializeKeypairField(T iroha::keypair_t::*field, const V &value) {
+  auto deserializeKeypairField(T keypair_t::*field, const V &value) {
     return [=](auto keypair) mutable {
-      return iroha::hexstringToArray<T::size()>(value)
-          | iroha::assignObjectField(keypair, field);
+      return hexstringToArray<T::size()>(value)
+          | assignObjectField(keypair, field);
+    };
+  }
+
+  std::string encrypt(const privkey_t &privkey,
+                      const std::string &pass_phrase) {
+    std::string ciphertext(privkey.size(), '\0');
+    const auto pass_size = std::max(1ul, pass_phrase.size());
+
+    for (auto i = 0u; i < privkey.size(); i++) {
+      ciphertext[i] = privkey[i] ^ pass_phrase[i % pass_size];
+    }
+    return ciphertext;
+  }
+
+  auto deserializedEncrypted(const std::string &s,
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
+      std::string key(binstr.size(), '\0');
+      const auto pass_size = std::max(1ul, pass_phrase.size());
+
+      for (auto i = 0u; i < binstr.size(); i++) {
+        key[i] = binstr[i] ^ pass_phrase[i % pass_size];
+      }
+      return nonstd::make_optional(key);
```

Comment:
why return optional here?

Reply:
For binding here:
https://github.com/hyperledger/iroha/blob/38862e33674375535eca98168c3a951e57ecdb10/libs/crypto/keys_manager_impl.cpp#L131-L134